### PR TITLE
CIVIL-475 - EventCollector Error Handling / Restart ; Polling Mode

### PR DESCRIPTION
**Pull request by petersng**
- Fixes the error handling in the EventCollector / watchers stack. Now errors cascade down to the `StartCollection` method where we can handle it properly.  We were not handling the errors from the watchers at all. This not only improves error handling, but allows us to capture the disconnect errors on the websocket subscriptions.
- Resets the ws connection on error in the EventCollector.
- Re-runs the filterer on every restart of the ws connection reset.
- Adds a polling mode to enable filterers only polling for events.  Enable this manually for now in the config, but can figure out some logic to automatically kick it into polling mode.  Noticed that Infura was having issues with subscriptions, this is a backup plan if it goes down again.

### CODE REVIEW

File: pkg/eventcollector/eventcollector.go [hunk 1]
```diff
@@ -213,41 +237,60 @@ func (c *EventCollector) StartChan() chan bool {
...
+				log.Errorf("Error publishing trigger message: err: %v", err)
+			}
+		}
 
-	c.sendStartSignal()
+		if c.pollingEnabled {
+			once.Do(func() { c.sendStartSignal() })
+			log.Infof("Polling enabled, waiting...")
+			select {
+			case <-time.After(time.Duration(c.pollingIntSecs()) * time.Second):
```
Comment (innainu) [inline pkg/eventcollector/eventcollector.go]: Trying to understand the code -- so where is the part that runs the filterer after each `c.pollingintsecs` duration?

Comment (petersng) [reply pkg/eventcollector/eventcollector.go]: yeah, i'll try and make this clearer here.

**Comment by petersng** [comment]
seems to be locking up during unsubscribing during some error restarts, so looking into it.

**Comment by petersng** [comment]
Solution for the lockups in #128

**Pull request merged**